pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/UnnecessaryImport.xml

```diff
@@ -1474,8 +1474,281 @@ public class Foo {
+                }
+            }
+        ]]></code>
+    </test-code>
+<!--    <test-code>-->
+<!--        <description>#5852: [java] UnnecessaryImport with multiple static imports from same class - true positive</description>-->
+<!--        <expected-problems>3</expected-problems>-->
+<!--        <expected-linenumbers>3,4,5</expected-linenumbers>-->
+<!--        <code><![CDATA[-->
+<!--            package io.quarkus.runtime.configuration;-->
+
+<!--            import static io.quarkus.runtime.ConfigConfig.Bar;-->
+<!--            import static io.quarkus.runtime.ConfigConfigg.Bar;-->
+<!--            import static io.quarkus.runtime.ConfigConfig.Foo;-->
+<!--            import static io.quarkus.runtime.ConfigConfig.Foo.fail;-->
+<!--            import static io.quarkus.runtime.ConfigConfig.Foo.warn;-->
+
+<!--            import java.util.Map;-->
+<!--            import org.jboss.logging.Logger;-->
+
+<!--            @Recorder-->
+<!--            public class ConfigRecorder {-->
+<!--                private static final Logger log = Logger.getLogger(ConfigRecorder.class);-->
+
+<!--                public void handleConfigChange(Map<String, ConfigValue> buildTimeRuntimeValues) {-->
+<!--                    if (!mismatches.isEmpty()) {-->
+<!--                        var msg = "Build time property cannot be changed at runtime:\n" + String.join("\n", mismatches);-->
+<!--                        var buildTimeMismatchAtRuntime = config-->
+<!--                                .getOptionalValue("quarkus.config.build-time-mismatch-at-runtime", String.class)-->
+<!--                                .orElse(warn);-->
+<!--                        if (fail.equals(buildTimeMismatchAtRuntime)) {-->
+<!--                            throw new IllegalStateException(msg);-->
+<!--                        } else if (warn.equals(buildTimeMismatchAtRuntime)) {-->
+<!--                            log.warn(msg);-->
+<!--                        }-->
+<!--                    }-->
+<!--                }-->
+<!--            }-->
+<!--        ]]></code>-->
+<!--    </test-code>-->
+
+
+    <!-- Test cases for static imports of nested classes -->
+    <test-code>
+        <description>Static import of nested class</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import static java.util.Map.Entry;
+public class Foo {
+    Entry<String, String> entry;
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>Static on-demand import of nested class</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import static java.util.Map.*;
+public class Foo {
+    Entry<String, String> entry;
+}
+        ]]></code>
+    </test-code>
+
+    <!-- Test cases for imports used in annotations -->
+    <test-code>
+        <description>Import used in annotation value</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import some.pkg.MyEnum;
```

Comment:
This test case should be improved: Please create the related classes/types in src/test/java in package `net.sourceforge.pmd.lang.java.rule.codestyle.unnecessaryimport` so that we actually use type resolution correctly.